components/atg/Terminal/index.tsx

```diff
@@ -0,0 +1,289 @@
+import {
+  ForwardedRef,
+  forwardRef,
+  useCallback,
+  useEffect,
+  useRef,
+  useState,
+} from "react";
+import { TerminalProps, TerminalCommands } from "./types"; // Assuming you have a types file where `TerminalCommands` is defined
+import CloseIcon from "@mui/icons-material/Close"; // Use CloseIcon to mimic the red button on Mac
+
+export interface AdditionalProps {
+  inputRef: React.RefObject<HTMLInputElement>;
+  terminalTheme: boolean;
```

Comment:
how is terminalTheme boolean ? shouldnt it be a string or is it named incorrectly

Reply:
terminalTheme is boolean. If it's true then it is light theme if it false it is dark mode. Let me correct the naming.